### PR TITLE
add page for LASTWITHTIME

### PR DESCRIPTION
**Pull request by matthewhelmke**

### CODE REVIEW

File: configuration-reference/functions/firstwithtime.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+description: This section contains reference documentation for the firstwithtime function.
+---
+
+# firstwithtime
+
+Get the first value of `dataColumn` where the `timeColumn` is used to define the time of `dataColumn` and the `dataType` specifies the type of `dataColumn`, which can be `BOOLEAN`, `INT`, `LONG`, `FLOAT`, `DOUBLE`, `STRING`
```
Comment (ksnijjer) [inline configuration-reference/functions/firstwithtime.md]: Looking at the [code](https://github.com/apache/pinot/blob/e08e66c3991b40f0d95c48b9436b97d406fa3311/pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FirstWithTimeAggregationFunction.java#L37C1-L47C4) first two arguments to the function are expressions and third is a literal specifying the return type of dataColumn. 
Help snippet from code:
`/**
 * This function is used for FirstWithTime calculations.
 * <p>The function can be used as FirstWithTime(dataExpression, timeExpression, 'dataType')
 * <p>Following arguments are supported:
 * <ul>
 *   <li>dataExpression: expression that contains the column to be calculated first on</li>
 *   <li>timeExpression: expression that contains the column to be used to decide which data is first, can be any
 *   Numeric column</li>
 *   <li>dataType: the data type of data column</li>
 * </ul>
 */`
 
 This seems more easier to understand definition vs current one.

File: configuration-reference/functions/firstwithtime.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Signature
+
+> FIRSTWITHTIME(dataColumn, timeColumn, 'dataType')
+
+## Usage Example
+
+This example is based on the [Streaming Quick Start](../../basics/getting-started/quick-start.md#streaming).
+
+```sql
+select FIRSTWITHTIME(group_name, __metadata$recordTimestamp, 'STRING')
```
Comment (ksnijjer) [inline configuration-reference/functions/firstwithtime.md:21]: Looking at defined schema column used for timeExpression is actually a string 
    {
      "dataType": "STRING",
      "name": "__metadata$recordTimestamp"
    }
and not numeric per code definition, however I am guessing this query runs fine? If yes then perhaps internally we do implicit String to numeric conversion but then more user relatable e.g could be just using time col. for the timeExpression `mTime`. Thoughts?

File: configuration-reference/functions/lastwithtime.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+description: This section contains reference documentation for the lastwithtime function.
+---
+
+# lastwithtime
+
+Get the last value of `dataColumn` where the `timeColumn` is used to define the time of `dataColumn` and the `dataType` specifies the type of `dataColumn`, which can be `BOOLEAN`, `INT`, `LONG`, `FLOAT`, `DOUBLE`, `STRING`
+
+## Signature
+
+> LASTWITHTIME(dataColumn, timeColumn, 'dataType')
```
Comment (ksnijjer) [inline configuration-reference/functions/lastwithtime.md:14]: Same comments as FirstWithTime()

File: configuration-reference/functions/firstwithtime.md (hunk 1 above)
Comment (Jackie-Jiang) [reply configuration-reference/functions/firstwithtime.md]: `timeColumn` can be of type `TIMESTAMP`, `INT`, `LONG`

Comment (Jackie-Jiang) [inline configuration-reference/functions/firstwithtime.md]: I feel it is a little bit more clear this way: Get the value of `dateColumn` with the smallest `timeColumn` value...

File: configuration-reference/functions/firstwithtime.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+description: This section contains reference documentation for the firstwithtime function.
+---
+
+# firstwithtime
+
+This function is used for FirstWithTime calculations.
```
Comment (kelseiv) [inline configuration-reference/functions/firstwithtime.md]:
```suggestion
```

File: configuration-reference/functions/firstwithtime.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+The function will return the value of `dataColumn` with the smallest `timeColumn` value where:
```
Comment (kelseiv) [inline configuration-reference/functions/firstwithtime.md]:
```suggestion
Returns the value of `dataColumn` with the smallest `timeColumn` value where:
```

File: configuration-reference/functions/firstwithtime.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+- `timeColumn` is used to define the time of `dataColumn`, which can be of type `TIMESTAMP`, `INT`, `LONG`
+- `dataType` specifies the type for `dataColumn`, which can be `BOOLEAN`, `INT`, `LONG`, `FLOAT`, `DOUBLE`, `STRING`
+
+## Signature
+
+> FIRSTWITHTIME(dataColumn, timeColumn, 'dataType')
+
+## Usage Example
```
Comment (kelseiv) [inline configuration-reference/functions/firstwithtime.md]:
```suggestion
## Example
```

File: configuration-reference/functions/lastwithtime.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+---
+description: This section contains reference documentation for the lastwithtime function.
+---
+
+# lastwithtime
+
+This function is used for LastWithTime calculations.
```
Comment (kelseiv) [inline configuration-reference/functions/lastwithtime.md]:
```suggestion
```

File: configuration-reference/functions/lastwithtime.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+The function will return the value of `dataColumn` with the largest `timeColumn` value where:
```
Comment (kelseiv) [inline configuration-reference/functions/lastwithtime.md]:
```suggestion
Returns the value of `dataColumn` with the largest `timeColumn` value where:
```

**Review by kelseiv — APPROVED** [review]

**Pull request merged**